commitments/da_service_op_commitment.go

```diff
@@ -0,0 +1,67 @@
+package commitments
+
+import (
+	"fmt"
+	"log"
+)
+
+type DAServiceOPCommitmentType byte
+
+const (
+	EigenDAByte DAServiceOPCommitmentType = 0
+)
+
+// DAServiceOPCommitment represents a value of one of two possible types (Keccak256Commitment or DAServiceCommitment).
+type DAServiceOPCommitment struct {
+	eigendaCommitment *EigenDACommitment
+}
+
+var _ Commitment = (*DAServiceOPCommitment)(nil)
+
+func EigenDAWrapperCommitment(value EigenDACommitment) DAServiceOPCommitment {
+	return DAServiceOPCommitment{eigendaCommitment: &value}
+}
+
+func (e DAServiceOPCommitment) IsEigenDA() bool {
+	return e.eigendaCommitment != nil
+}
+
+func (e DAServiceOPCommitment) MustEigenDAValue() EigenDACommitment {
+	if e.eigendaCommitment != nil {
+		return *e.eigendaCommitment
+	}
+	log.Panic("CommitmentEither does not contain a Keccak256Commitment value")
```

Comment:
Why are we panicking? Should prolly be handling errors gracefully for client context

Reply:
This function should only be called when preceded by a call to `.IsEigenDACommitment()`. Prefixing the function name with `Must` is an idiom in golang to indicate that the function will panic if not called in the right context.

IMO returning an error as well is clunky and redundant with the existence of the IsEigenDACommitment() function.

Reply:
so my issue with this is that we'll panic on the server handler routine which will result in a dropped connection with insufficient information being provided to the proxy client

Reply:
The pattern I'm advocating for:
```
var comm DAServiceOPCommitment
comm.Unmarshal(b)
if comm.IsEigenDA() {
    eigendaComm := comm.MustEigenDAValue()
    // do something
}
```

The below pattern is is everywhere in golang code, is not particularly, and is isomorphic to the above:
```
foo := getObjectOrNil()
if foo != nil {
    foo.bar() // possible NPE if we didn't make the surrounding check
}
```